ros_babel_fish/test/message.cpp

```diff
@@ -691,6 +691,142 @@ TEST( MessageTest, arrayMessage )
+  ASSERT_EQ( (isCompatible<uint16_t, double>()), true );
+  ASSERT_EQ( (isCompatible<uint32_t, uint8_t>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, uint16_t>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, uint32_t>()), true );
+  ASSERT_EQ( (isCompatible<uint32_t, uint64_t>()), true );
+  ASSERT_EQ( (isCompatible<uint32_t, int8_t>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, int16_t>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, int32_t>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, int64_t>()), true );
+  ASSERT_EQ( (isCompatible<uint32_t, float>()), false );
+  ASSERT_EQ( (isCompatible<uint32_t, double>()), true );
+  ASSERT_EQ( (isCompatible<uint64_t, uint8_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, uint16_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, uint32_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, uint64_t>()), true );
+  ASSERT_EQ( (isCompatible<uint64_t, int8_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, int16_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, int32_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, int64_t>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, float>()), false );
+  ASSERT_EQ( (isCompatible<uint64_t, double>()), false );
+  ASSERT_EQ( (isCompatible<int8_t, uint8_t>()), false );
+  ASSERT_EQ( (isCompatible<int8_t, uint16_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, uint32_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, uint64_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, int8_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, int16_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, int32_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, int64_t>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, float>()), true );
+  ASSERT_EQ( (isCompatible<int8_t, double>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, uint8_t>()), false );
+  ASSERT_EQ( (isCompatible<int16_t, uint16_t>()), false );
+  ASSERT_EQ( (isCompatible<int16_t, uint32_t>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, uint64_t>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, int8_t>()), false );
+  ASSERT_EQ( (isCompatible<int16_t, int16_t>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, int32_t>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, int64_t>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, float>()), true );
+  ASSERT_EQ( (isCompatible<int16_t, double>()), true );
+  ASSERT_EQ( (isCompatible<int32_t, uint8_t>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, uint16_t>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, uint32_t>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, uint64_t>()), true );
+  ASSERT_EQ( (isCompatible<int32_t, int8_t>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, int16_t>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, int32_t>()), true );
+  ASSERT_EQ( (isCompatible<int32_t, int64_t>()), true );
+  ASSERT_EQ( (isCompatible<int32_t, float>()), false );
+  ASSERT_EQ( (isCompatible<int32_t, double>()), true );
```

Comment:
I'm not sure if I agree that an assignment of (unsigned) integer types larger than 2 bytes to floating-point types shouldn't be possible (without warning).
The introduced error for the values that a float can not represent exactly is negligible in most cases IMO and if your message member type is floating-point, you have to expect some floating-point error.

Reply:
On second thought, yes, I agree. That was too stringent and warning for conversions to floating point is an overkill. If the user asks for a float then that's most likely the precision they are happy with. I updated the code and tests.